yt_dlp/extractor/duoplay.py

```diff
@@ -0,0 +1,118 @@
+            'thumbnail': r're:https://.+\.jpg(?:\?c=\d+)?$',
+            'description': 'md5:8ef98f38569d6b8b78f3d350ccc6ade8',
+            'upload_date': '20170523',
+            'timestamp': 1495567800,
+            'series': 'Siberi võmm',
+            'series_id': '4312',
+            'season': 'Season 2',
+            'season_number': 2,
+            'episode': 'Operatsioon "Öö"',
+            'episode_number': 12,
+            'episode_id': 24,
+        },
+    }, {
+        'note': 'Empty title',
+        'url': 'https://duoplay.ee/17/uhikarotid?ep=14',
+        'md5': '6aca68be71112314738dd17cced7f8bf',
+        'info_dict': {
+            'id': '17_14',
+            'ext': 'mp4',
+            'title': 'Ühikarotid',
+            'thumbnail': r're:https://.+\.jpg(?:\?c=\d+)?$',
+            'description': 'md5:4719b418e058c209def41d48b601276e',
+            'upload_date': '20100916',
+            'timestamp': 1284661800,
+            'series': 'Ühikarotid',
+            'series_id': '17',
+            'season': 'Season 2',
+            'season_number': 2,
+            'episode_id': 14,
+            'release_year': 2010,
+        },
+    }, {
+        'note': 'Movie',
+        'url': 'https://duoplay.ee/4325/naljamangud',
+        'md5': '2b0bcac4159a08b1844c2bfde06b1199',
+        'info_dict': {
+            'id': '4325',
+            'ext': 'mp4',
+            'title': 'Näljamängud',
+            'thumbnail': r're:https://.+\.jpg(?:\?c=\d+)?$',
+            'description': 'md5:fb35f5eb2ff46cdb82e4d5fbe7b49a13',
+            'cast': ['Jennifer Lawrence', 'Josh Hutcherson', 'Liam Hemsworth'],
+            'upload_date': '20231109',
+            'timestamp': 1699552800,
+            'release_year': 2012,
+        },
+    }, {
+        'note': 'Movie without expiry',
+        'url': 'https://duoplay.ee/5501/pilvede-all.-neljas-ode',
+        'md5': '7abf63d773a49ef7c39f2c127842b8fd',
+        'info_dict': {
+            'id': '5501',
+            'ext': 'mp4',
+            'title': 'Pilvede all. Neljas õde',
+            'thumbnail': r're:https://.+\.jpg(?:\?c=\d+)?$',
+            'description': 'md5:d86a70f8f31e82c369d4d4f4c79b1279',
+            'cast': ['Inga Lunge', 'Elisabet Reinsalu', 'Liis Haab', 'Rita Rätsepp', 'Ain Lutsepp', 'Indrek Sammul', 'Piret Rauk', 'Karin Tammaru', 'Guido Kangur'],
+            'upload_date': '20221214',
+            'timestamp': 1671054000,
+            'release_year': 2018,
+        },
+    }]
+
+    def _real_extract(self, url):
+        telecast_id, episode = self._match_valid_url(url).group('id', 'ep')
+        video_id = join_nonempty(telecast_id, episode, delim='_')
+        webpage = self._download_webpage(url, video_id)
+        video_player = try_call(lambda: extract_attributes(
+            get_element_text_and_html_by_tag('video-player', webpage)[1]))
+        if not video_player or not video_player.get('manifest-url'):
+            self.raise_no_formats('No video found', expected=True)
```

Comment:
better to raise ExtractorError here to prevent ugliness when `--ignore-no-formats` is passed
```suggestion
            raise ExtractorError('No video found', expected=True)
```